onnx/defs/tensor/defs.cc

```diff
@@ -3646,37 +3610,46 @@ ONNX_OPERATOR_SET_SCHEMA(
         })
         .SetContextDependentFunctionBodyBuilder(
             [](const FunctionBodyBuildContext& ctx, const OpSchema& schema, FunctionProto& functionProto) {
-              const auto& input_shape = ctx.getInputType(0)->tensor_type().shape();
-              const int64_t input_rank = input_shape.dim_size();
+              FunctionBuilder builder(functionProto);
+              builder.Const("k2", std::vector<int64_t>{2});
 
               auto axes_attr = ctx.getAttribute("axes");
               std::vector<int64_t> axes;
-              if (axes_attr) {
+              if (axes_attr) {  // axes provided, need to work on a subset of dimensions
                 axes = RetrieveValues<int64_t>(*axes_attr);
-              } else {
-                axes.resize(input_rank);
-                std::iota(axes.begin(), axes.end(), 0);
+                builder.Const("axes_input", axes);
```

Comment:
I don't see `axes` being initialized above. Either it needs to be defined, or a better option is to use
```cpp
    builder.Add("axes_input = Constant <value_ints : ints = @axes>()");
```

Reply:
I don't understand this comment. `axes` is declared in line 3617 and the values are read in line 3619.
That being said, your suggestion seems better. I'll change it.

Reply:
Done